docs/src/model_developer/market_bid_cost.md

```diff
@@ -0,0 +1,96 @@
+# PowerSystems MarketBidCost example
+
+## MarketBidCost 
+Is a Operational Cost  data structure that allows the user to run a production cost model that is very similar to most US electricity market auctions with bids for energy and ancilliary services.
+```julia
+    mutable struct MarketBidCost <: OperationalCost
+        variable::Union{Nothing, IS.TimeSeriesKey}
+        no_load::Float64
+        start_up::NamedTuple{(:hot, :warm, :cold), NTuple{3, Float64}}
+        shut_down::Float64
+        ancillary_services::Vector{Service}
+    end
+```
+
+#### Arguments
+- `variable::Union{Nothing, IS.TimeSeriesKey}`: Variable Cost TimeSeriesKey
+- `no_load::Float64`: minimum operating cost
+- `start_up::NamedTuple{(:hot, :warm, :cold), NTuple{3, Float64}}`: start-up cost at different stages of the thermal cycle. Warm is also refered as intermediate in some markets
+- `shut_down::Float64`: shut-down cost, validation range: `(0, nothing)`, action if invalid: `warn`
+- `ancillary_services::Vector{Service}`: Bids for the ancillary services
+
+## Adding Energy bids to MarketBidCost
+
+#### Step 1: Constructiong device with MarketBidCost
+When using MarketBidCost, the user can add the cost struct to the , at this point the actual cost bids aren't expected to be populated/passed. The code below shows how we can create a thermal device with MarketBidCost.
+
+```julia
+generator = ThermalStandard(
+        name = "Brighton",
+        available = true,
+        status = true,
+        bus = nodes5[5],
+        active_power = 6.0,
+        reactive_power = 1.50,
+        rating = 0.75,
+        prime_mover = PrimeMovers.ST,
+        fuel = ThermalFuels.COAL,
+        active_power_limits = (min = 0.0, max = 6.0),
+        reactive_power_limits = (min = -4.50, max = 4.50),
+        time_limits = (up = 0.015, down = 0.015),
+        ramp_limits = (up = 5.0, down = 3.0),
+        operation_cost = MarketBidCost(
+            nothing, 
+            0.0, 
+            (hot = 393.28, warm = 455.37, cold = 703.76),, 
+            0.75, 
+            Vector{Service}()
+            ),
+        base_power = 100.0,
+    ),
+
+```
+OR
+```julia
+generator = ThermalStandard(nothing)
```

Comment:
We shouldn't encourage people to use this constructor. this is for demonstration purposes.